cmd/main.go

```diff
@@ -186,3 +188,23 @@ func main() {
 func virtLauncherSelector() labels.Selector {
 	return labels.SelectorFromSet(map[string]string{virtv1.AppLabel: "virt-launcher"})
 }
+
+func pruneIrrelevantPodData(obj interface{}) (interface{}, error) {
+	oldPod, ok := obj.(*corev1.Pod)
+	if !ok {
+		return obj, nil
+	}
```

Comment:
i am not familiar with Transform functions by heart
should it return error here ? or it is fine as you did because it can catch other types and if the cast failed it should just ignore and continue ?

Reply:
This is common for all types which we use in controller runtime (pods / vmis / vms).

I just want to act on pods.